### ISSUE TITLE
makeCpGregions reduced a huge amount of CpGs to be tested

### ISSUE THREAD

**Issue by ggzheng** [christpa/DMRScan#14]
Hi,

I am having an issue with applying the region function will keeping the enough number of CpGs. Before putting tValues into the makeCpGregions function, I have 801812 CpGs. After applying this function, I only have 13484 left to be tested! And this further caused an other issue with sampling for flexible windows.

R:

length(tValues) #[1] 801812
#use default value: maxGap = 500, minCpG = 2
regions <- makeCpGregions(observations=tValues, chr=pos2[,1], pos=pos2[,2])
length(regions) #3032
n.CpG <- sum(sapply(regions, length)) 
n.CpG #13484

**Comment by christpa** [comment]
Hi Zheng, 

Try to increase the maxGap to about 1000 and see if that helps. 
Also check that you have the updated code and that the regions are on different chromosomes. In a previous bug, the makeCpGregions() only reported the regions for the first chromosome. 

Best
Christian

**Comment by ggzheng** [comment]
Yes, I checked it only output the regions for the first chromosome. 

I updated the package again by using BiocManager::install("DMRScan") and updated all corresponding packages needed. However, this still output the same results.

BTW, BiocManager::install("DMRScan") suggested that it tried URL 'https://bioconductor.org/packages/3.10/bioc/src/contrib/DMRScan_1.11.0.tar.gz'

**Comment by christpa** [comment]
This should have been fixed and pushed to bioconductor. Are you using version 1.10.0?

**Comment by christpa** [comment]
Try to run this code: 

```{r}
data(DMRScan.methylationData)
data(DMRScan.phenotypes) 

test.statistics <- apply(DMRScan.methylationData,1,function(x,y) summary(glm(y ~ x, family = binomial(link = "logit")))$coefficients[2,3],   y = DMRScan.phenotypes)

positions <- data.frame(matrix(as.integer(unlist(strsplit(names(test.statistics), split="chr|[.]"))), ncol = 3, byrow = TRUE))[,-1]

## Update positions to include 2 separate chromosomes
positions$X2 <- c(21,rep(21:22, each = nrow(positions)/2))

min.cpg <- 4 
max.gap <- 750  
     
regions <- makeCpGregions(observations = test.statistics, chr = positions[,1], pos = positions[,2], maxGap = max.gap, minCpG = min.cpg)

```

Then `regions[[1]]` should be on chromosome 21 and `regions[[1000]]` should be on chromosome 22.

**Comment by ggzheng** [comment]
I tested your code, and the output gave results both for chromosome 21 and 22. 

I actually tested it similarly in my data by reducing the data only with SNPs on 2 picked chromosomes. Interestingly, the code worked properly if I picked "21 + 22" or "15+16", but the code failed again and only reported the first chromosome if picked "1+2" or "2+3". Something I noticed that, "21 + 22" and "15+16" reported much fewer total number of regions in 2 CHRs than that from "1+2" and "2+3" where only reporting 1 CHR. I am not sure if this is related.

**Comment by christpa** [comment]
This is very odd. 
When I substitute the `positions[,1] <- sort(sample(1:5, nrow(positions),T))` it gives roughly the same number of regions. 

How is your DNA methylation data structured, is it sequencing or chip data (450k, EPIC)?

**Comment by ggzheng** [comment]
Sorry I missed your comments. My data is chip data. 

It is really wired. To provide you some numbers, I tested again. Please see below. 

When I used 

1) CHR 21,22, it output regions from both CHRs and the total number of regions is 1012
2) CHR 19,20,21 it only output regions from CHR 19, and the total number of regions is 987
